src/main/java/com/elixr/poc/bulkimport/service/reader/FileReader.java

```diff
@@ -0,0 +1,165 @@
+                FileOperationEnum.PATIENT_FIRSTNAME.getFileKey(), FileOperationEnum.PATIENT_LASTNAME.getFileKey(),
+                FileOperationEnum.PATIENT_AGE.getFileKey(), FileOperationEnum.PATIENT_ADDRESS.getFileKey(),
+                FileOperationEnum.DOCTOR_ID.getFileKey()};
+        HashMap<String, String> columnHeader = new HashMap<>();
+        try {
+            XSSFWorkbook workbook = new XSSFWorkbook(inputStream);
+            XSSFSheet sheet = workbook.getSheetAt(0);
+            Row row;
+            Cell cell;
+            List<RowResponse> rowResponse = new ArrayList<>();
+            Iterator rowIterator = sheet.rowIterator();
+            while (rowIterator.hasNext()) {
+                row = (XSSFRow) rowIterator.next();
+                if (row.getRowNum() != 0) {
+                    int headerCount = 0;
+                    for (int cellNumber = 0; cellNumber < row.getLastCellNum(); cellNumber++) {
+                        cell = row.getCell(cellNumber, Row.MissingCellPolicy.CREATE_NULL_AS_BLANK);
+                        stringValue = checkCellType(cell);
+                        columnHeader.put(Arrays.stream(patientKey).toList().get(headerCount), stringValue);
+                        headerCount++;
+                    }
+                    Patient patient;
+                    if (StringUtils.isEmpty(columnHeader.get(FileOperationEnum.PATIENT_AGE.getFileKey()))) {
+                        patient = buildPatientWhenAgeIsEmpty(columnHeader);
+                    } else {
+                        patient = buildPatient(columnHeader);
+                    }
+                    rowResponse.add(performAction(patient, columnHeader, row.getRowNum() + 1));
+                }
+            }
+            return GenericResponse.builder().status(FileOperationEnum.SUCCESS.getFileKey()).data(rowResponse).build();
+        } catch (Exception exception) {
+            return GenericResponse.builder().status(FileOperationEnum.FAILURE.getFileKey()).data(null).build();
+        }
+    }
+
+    /**
+     * Check cell type like is it String or Numeric.
+     * @param cell
+     * @return
+     */
+    private String checkCellType(Cell cell) {
+        if (CellType.STRING == cell.getCellType() || CellType.BLANK == cell.getCellType()) {
+            stringValue = cell.getStringCellValue();
+        } else if (CellType.NUMERIC == cell.getCellType() || CellType.BLANK == cell.getCellType()) {
+            stringValue = String.valueOf((int) cell.getNumericCellValue());
+        }
+        return stringValue;
+    }
+
+    /**
+     * Building the Patient object from columnHeader.
+     * Calling the respective methods based on the action.
+     *
+     * @param columnHeader
+     * @return
+     */
+    private Patient buildPatient(HashMap<String, String> columnHeader) {
+        String doctorId = columnHeader.get(FileOperationEnum.DOCTOR_ID.getFileKey());
```

Comment:
The same question which I asked previously on why you are adding it into a list instead of a string applies here too

Reply:
To store multiple Doctors to an single patient.
To get many to many relation between patient and doctor.
Need to work on reading, comma separated values for doctor column.